R/data.table.R

```diff
@@ -1019,9 +1019,12 @@ replace_dot_alias = function(e) {
             # .SDcols is of the format a:b, ensure none of : arguments is a call data.table(V1=-1L, V2=-2L, V3=-3L)[,.SD,.SDcols=-V2:-V1] #4231
             .SDcols = eval(colsub, setattr(as.list(seq_along(x)), 'names', names_x), parent.frame())
           } else {
-            if (colsub %iscall% 'patterns') {
-              # each pattern gives a new filter condition, intersect the end result
-              .SDcols = Reduce(intersect, eval_with_cols(colsub, names_x))
+            if (is.call(colsub) && !is.null(list_or_vector <- eval_with_cols(colsub, names_x))) {
```

Comment:
is this change related to the current PR?

Reply:
it was required to support user-defined functions but I reverted the change